onedal/spmd/neighbors/neighbors.py

```diff
@@ -0,0 +1,43 @@
+#===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+from abc import ABC
+from ...common._spmd_policy import _get_spmd_policy
+from onedal.neighbors import KNeighborsClassifier as KNeighborsClassifier_Batch
+from onedal.neighbors import KNeighborsRegressor as KNeighborsRegressor_Batch
+
+
+class NeighborsCommonBaseSPMD(ABC):
+    def _get_infer_policy(self, queue, *data):
+        return _get_spmd_policy(queue)
+
+
+class NeighborsBaseSPMD(NeighborsCommonBaseSPMD, ABC):
+    pass
+
+
+class KNeighborsClassifier(NeighborsBaseSPMD, KNeighborsClassifier_Batch):
+    def predict_proba(self, X, queue=None):
+        raise NotImplementedError("predict_proba not supported in distributed mode.")
+
+
+class KNeighborsRegressor(NeighborsBaseSPMD, KNeighborsRegressor_Batch):
+    def predict(self, X, queue=None):
+        return self._predict_gpu(X, queue=queue)
+
+
+class NearestNeighbors(NeighborsBaseSPMD):
+    pass
```

Comment:
If we don't support SPMD, there is no need adding it here

Reply:
I think it should be left since it uses NeighborsBaseSPMD/NeighborsCommonBaseSPMD

Reply:
We don't need `NeighborsBaseSPMD` as well here.

Reply:
Remove it, please. Leave only NeighborsCommonBaseSPMD, the code where you really changing behavior. Remove `NearestNeighbors` from init file as well. If we don't support SPMD interface, no need to put this class here.

Reply:
We do support it. It's the same circumstance as in [linear regression](https://github.com/intel/scikit-learn-intelex/blob/master/onedal/spmd/linear_model/linear_model.py#L27). No extensions of the class itself are introduced but there are extensions to classes it inherits from, so I believe it is necessary here.

Reply:
There is no need for inheriting `NeighborsBaseSPMD`. This are not public. there is no need for such `NeighborsBaseSPMD`